### ISSUE TITLE
Inference Test Slower than Reported

### ISSUE THREAD

**Issue by LeoCyclope** [Megvii-BaseDetection/YOLOX#622]
Hi guys :) 

I trained a YoloV3 (YOLOX-Darknet53) and used the corresponding experiments except. that I. have changed the data set and reduced the img size to 416x416.

I am using a V100 not used when testing and still the inference time is slower than reported:

Both GPU and CPU are free.

I launched it with: 
`python tools/eval.py -f YOLOX/exps/default/yolov3_dai.py -c YOLOX/YOLOX_outputs/YoloV3-V1/best_ckpt.pth -b 1 -d 1 --conf 0.001 --fp16 --fuse`

**Average forward time: 16.57 ms, Average NMS time: 0.79 ms, Average inference time: 17.36 ms**

Any idea on where the reason can come from?

For information, this is my experiment file:

`#!/usr/bin/env python3
# -*- coding:utf-8 -*-
# Copyright (c) Megvii, Inc. and its affiliates.

import os
import torch
import torch.nn as nn
import torch.distributed as dist

from yolox.exp import Exp as MyExp

class Exp(MyExp):
    def __init__(self):
        super(Exp, self).__init__()
        self.depth = 1.0
        self.width = 1.0
        self.exp_name = os.path.split(os.path.realpath(__file__))[1].split(".")[0]
        
        # --------------  training config --------------------- #
        self.warmup_epochs = 5
        self.max_epoch = 300
        
        # ---------------- model config ---------------- #
        self.num_classes = 3
        self.depth = 1.00
        self.width = 1.00

        # ---------------- dataloader config ---------------- #
        # set worker to 4 for shorter dataloader init time
        self.data_num_workers = 4
        self.input_size = (416, 416)
        self.val_ann = "dai_detector_test.json"
        self.train_ann = "dai_detector_train.json"
        
        self.data_dir = "datasets/DAI_Detector/"
        
        self.exp_name = os.path.split(os.path.realpath(__file__))[1].split(".")[0]
        
        # -----------------  testing config ------------------ #
        self.test_size = (416, 416)
        self.test_conf = 0.01
        self.nmsthre = 0.45

    def get_model(self, sublinear=False):
        def init_yolo(M):
            for m in M.modules():
                if isinstance(m, nn.BatchNorm2d):
                    m.eps = 1e-3
                    m.momentum = 0.03
        if "model" not in self.__dict__:
            from yolox.models import YOLOX, YOLOFPN, YOLOXHead
            backbone = YOLOFPN()
            head = YOLOXHead(self.num_classes, self.width, in_channels=[128, 256, 512], act="lrelu")
            self.model = YOLOX(backbone, head)
        self.model.apply(init_yolo)
        self.model.head.initialize_biases(1e-2)

        return self.model

    def get_data_loader(
            self, batch_size, is_distributed, no_aug=False, cache_img=False
        ):        
        
        from yolox.data import (
            YOLO_Darknet_COCO,
            TrainTransform,
            YoloBatchSampler,
            DataLoader,
            InfiniteSampler,
            MosaicDetection,
            worker_init_reset_seed,
        )
        
        from yolox.utils import (
            wait_for_the_master,
            get_local_rank,
        )

        
        
        local_rank = get_local_rank()

        with wait_for_the_master(local_rank):
            dataset = YOLO_Darknet_COCO(
                data_dir=self.data_dir,
                json_file=self.train_ann,
                img_size=self.input_size,
                name="train",
                preproc=TrainTransform(max_labels=50),
                cache=cache_img,
            )

        dataset = MosaicDetection(
            dataset,
            mosaic=not no_aug,
            img_size=self.input_size,
            preproc=TrainTransform(max_labels=120),
            degrees=self.degrees,
            translate=self.translate,
            mosaic_scale=self.mosaic_scale,
            mixup_scale=self.mixup_scale,
            shear=self.shear,
            perspective=self.perspective,
            enable_mixup=self.enable_mixup,
            mosaic_prob=self.mosaic_prob,
            mixup_prob=self.mixup_prob,
        )

        self.dataset = dataset

        if is_distributed:
            batch_size = batch_size // dist.get_world_size()

            
        sampler = InfiniteSampler(len(self.dataset), seed=self.seed if self.seed else 0)

        batch_sampler = YoloBatchSampler(
            sampler=sampler,
            batch_size=batch_size,
            drop_last=False,
            mosaic=not no_aug,
        )

        dataloader_kwargs = {"num_workers": self.data_num_workers, "pin_memory": True}
        dataloader_kwargs["batch_sampler"] = batch_sampler
        
        dataloader_kwargs["worker_init_fn"] = worker_init_reset_seed
        
        train_loader = DataLoader(self.dataset, **dataloader_kwargs)

        return train_loader
    
    def get_eval_loader(self, batch_size, is_distributed, testdev=False, legacy=False):

        from yolox.data import YOLO_Darknet_COCO, ValTransform

        valdataset = YOLO_Darknet_COCO(
            data_dir=self.data_dir,
            json_file=self.val_ann,
            img_size=self.input_size,
            name="val",
            preproc=ValTransform(legacy=legacy),
        )
       

        if is_distributed:
            batch_size = batch_size // dist.get_world_size()
            sampler = torch.utils.data.distributed.DistributedSampler(
                valdataset, shuffle=False
            )
        else:
            sampler = torch.utils.data.SequentialSampler(valdataset)

        dataloader_kwargs = {
            "num_workers": self.data_num_workers,
            "pin_memory": True,
            "sampler": sampler,
        }
        dataloader_kwargs["batch_size"] = batch_size
        val_loader = torch.utils.data.DataLoader(valdataset, **dataloader_kwargs)

        return val_loader
`

**Comment by ruinmessi** [comment]
Someone has checked the speed for yolox-l model (See #597 ). So I suggest you test a yolox-l model to verify your env first. 
You don't need re-train a new model. just add `--speed` to test the speed.

**Comment by LeoCyclope** [comment]
I tested with COCO and the yolox-l and I get:

**Average forward time: 23.24 ms, Average NMS time: 0.81 ms, Average inference time: 24.05 ms**

For information, I have:
Driver Version: 460.91.03    CUDA Version: 11.2
Pytorch: 1.9.0+cu111
Using conda environement with conda 4.10.1 and Python 3.7.10 

Any ideas?

THE GPU is free as well as the machine.

Thanks in advance,

Regards
